[new product] contour (#5308)
captn3m0/endoflife.date: products/contour.md

---

> title: contour

Reviewer: ```suggestion
title: Contour
```

> category: app

Reviewer: ```suggestion
category: server-app
```

> tags: cncf http-proxy kubernetes ingress-controller

Reviewer: As mentioned in /contribute, please do not add any new tags.

```suggestion
tags: cncf kubernetes
```

permalink: /contour
alternate_urls:

> -   /contour

Reviewer: Same as permalink, not needed.
```suggestion
```

releasePolicyLink: https://projectcontour.io/resources/support/

> changelogTemplate: https://github.com/projectcontour/contour/releases/tag/v1.29.1/v__LATEST__

Reviewer: ```suggestion
changelogTemplate: https://github.com/projectcontour/contour/releases/tag/v__LATEST__
```

releaseDateColumn: true
eolColumn: Bug and Security Fixes

auto:
  methods:
  -   git: https://github.com/projectcontour/contour.git

identifiers:

> -   repology: contour

Reviewer: https://repology.org/projects/contour/

Is there any distro packaging contour? The repology identifier looks wrong.

```suggestion
```

> -   purl: pkg:docker/projectcontour/contour

Reviewer: GHCR image based on https://projectcontour.io/quickstart/contour.yaml

```suggestion
-   purl: pkg:github/projectcontour/contour
-   purl: pkg:docker/projectcontour/contour
-   purl: pkg:docker/bitnami/contour
-   purl: pkg:oci/contour?repository_url=ghcr.io/projectcontour
```

> 

Reviewer: Lets add a comment for maintenance.
```suggestion
# eol(X) = releaseDate(X+3)
```

releases:
-   releaseCycle: "1.29"
    releaseDate: 2024-05-08
    eol: false
    latest: "1.29.1"
    latestReleaseDate: 2024-06-13

-   releaseCycle: "1.28"
    releaseDate: 2024-02-12
    eol: false
    latest: "1.28.5"
    latestReleaseDate: 2024-06-13

-   releaseCycle: "1.27"
    releaseDate: 2023-10-30
    eol: false
    latest: "1.27.4"
    latestReleaseDate: 2024-06-13

---

> > [contour](https://projectcontour.io/) is an ingress controller for Kubernetes that works by deploying the Envoy proxy as a reverse proxy and load balancer. Contour supports dynamic configuration updates out of the box while maintaining a lightweight profile.

Reviewer: title case, link to Envoy

```suggestion
> [Contour](https://projectcontour.io/) is an ingress controller for Kubernetes that works by deploying [Envoy](https://www.envoyproxy.io/) as a reverse proxy and load balancer. Contour supports dynamic configuration updates out of the box while maintaining a lightweight profile.
```

> Contour does quarterly releases and three supported releases. The first Contour version covered by the quarterly release cadence is Contour v1.20, released in Jan 2022. To understand better, for example, when Contour 1.30 releases, Contour 1.27 will fall out of support. In short, “supported” means that Contour will issue fixes for security and other critical bugs for that release’s supported lifetime. However, the project will require users to upgrade to the most recent patch release for a version to be supported.
> 
> That is:
> * The latest patch version in each release is the supported version.
> * If you are not running the supported version from your release train, you’ll need to upgrade first if you have any problems.
> * When a new patch is cut, that will become the supported version for that release.
> 
> [SemVer](https://semver.org/) is used for all packages released.

Reviewer: Explanation for "please run the latest patch" is already covered in the last sentence: "You should be running one of the supported release numbers listed above in the rightmost column.".

Makes the explanation a bit more concise and simpler to understand:

```suggestion
Contour follows [Semantic Versioning](https://semver.org/). A new minor version is released approximately once a quarter.

The latest three releases are supported with critical and security fixes.

A [compatibility matrix](https://projectcontour.io/resources/compatibility-matrix/) is published
for compatibility with the Envoy/Kubernetes versions supported for every release.
```
